org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/wizards/buildpaths/LibrariesWorkbookPage.java

```diff
@@ -527,7 +536,9 @@ private void libaryPageCustomButtonPressed(DialogField field, int index) {
 
 				fLibrariesList.setElements(elements);
```

Comment:
Note: this class is "guilty" to send too much updates, as it calls `fLibrariesList.removeAllElements();`above  **and** `fLibrariesList.setElements(elements);` here. What we only need is the last call, also `fLibrariesList.refresh();` is not needed as it is done by the list anyway.

The patch to apply on top of your commit is:
```
diff --git a/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/wizards/buildpaths/LibrariesWorkbookPage.java b/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/wizards/buildpaths/LibrariesWorkbookPage.java
index 871a51e..6a98fe1 100644
--- a/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/wizards/buildpaths/LibrariesWorkbookPage.java
+++ b/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/wizards/buildpaths/LibrariesWorkbookPage.java
@@ -518,3 +518,2 @@
 				boolean isModuleRootExpanded= getRootExpansionState(fLibrariesList, false);
-				fLibrariesList.removeAllElements();
 				RootCPListElement selectedCPElement= (RootCPListElement) selectedElements.get(0);
@@ -537,3 +536,2 @@
 				fLibrariesList.setElements(elements);
-				fLibrariesList.refresh();
 				if(fLibrariesList.getTreeViewer() != null) {
``'